ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/VulnerabilityStateTabs.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { CSSProperties } from 'react';
+import { Tabs, Tab, TabTitleText } from '@patternfly/react-core';
+
+import { vulnerabilityStates } from 'types/cve.proto';
+import useFeatureFlags from 'hooks/useFeatureFlags';
+import useURLStringUnion from 'hooks/useURLStringUnion';
+
+export type VulnerabilityStateTabsProps = {
+    className?: string;
+    titleOverrides?: {
+        observed?: string;
+        deferred?: string;
+        falsePositive?: string;
+    };
+};
+
+function VulnerabilityStateTabs({ className, titleOverrides }: VulnerabilityStateTabsProps) {
+    const { isFeatureFlagEnabled } = useFeatureFlags();
+    const isUnifiedDeferralsEnabled = isFeatureFlagEnabled('ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL');
+    const [vulnerabilityStateKey, setVulnerabilityStateKey] = useURLStringUnion(
+        'vulnerabilityState',
+        vulnerabilityStates
+    );
+
+    return isUnifiedDeferralsEnabled ? (
+        <Tabs
+            className={className}
+            activeKey={vulnerabilityStateKey}
+            onSelect={(_e, tab) => setVulnerabilityStateKey(tab)}
+            component="nav"
+            style={
+                {
+                    '--pf-c-tabs__link--BackgroundColor': 'var(--pf-global--BackgroundColor--100)',
```

Comment:
* Seeing the background color here raises a question whether color via `className` prop is independent or if component needs to encapsulate both for consistency?
* To reduce number of comments, `pf-u-pl-lg` seems like a detail of page layout better kept out of tabs component. Is there a way that `PageSection` element can take responsibility?

Reply:
The background color here is set to override a CSS var for the child tab components. Setting the background color via `className` would apply it to the tab container, which would spill out into an area we want to remain transparent. Encapsulating felt more straightforward than either creating a `tabClassName` prop to pass the relevent class from above, or composing in some other way.
![image](https://github.com/stackrox/stackrox/assets/1292638/7b586a31-18b7-4411-b8b9-b7ee39c9c619)
![image](https://github.com/stackrox/stackrox/assets/1292638/46ff440f-d885-4c9d-ac72-10e4ac6dab8e)

Agreed on the second comment, I'll see if I can more cleanly delegate layout to a parent in the one case that this is needed.

Reply:
On the first point, I was not clear enough. I did not mean `className` versus `style` prop, but instead whether `pf-u-background-color-100` in `className` prop is an independent choice depending on rendering context, or also inherently part of color theme specified within the component.

Reply:
For the second, is it possible that vertical flex layout might work for tabs in the `PageSection` element that has heading, blurb, and **Manage watched images** button?

Reply:
Aha, yes that makes sense. The `pf-u-background-color-100` is independent depending on the context. I think with your second suggestion I can update this to remove the need for a `className` prop altogether on the new tab component.